_episodes/08-vector-open-shapefile.md

```diff
@@ -75,101 +70,121 @@ Key metadata for all shapefiles include:
 
 ![Extent image](../fig/dc-spatial-vector/spatial_extent.png)
 
 We can convert these coordinates to a bounding box or acquire the index of the dataframe to access the geometry. Either of these polygons can be used to clip rasters (more on that later). 
 
-## Reading a Shapefile from a csv
+## Cropping a Shapefile
+Our `cropfield` dataset is rather large, containing data for the entirety of the European portion of the Netherlands. Before plotting it we will first select a specific section of to be our area of interest.
+
+We can create a cropped version of our dataset as follows:
+```python
+# Define a Boundingbox in RD
+xmin, xmax = (120000, 135000)
+ymin, ymax = (485000, 500000)
+cropfield_crop = cropfield.cx[xmin:xmax, ymin:ymax]
+```
 
-So far we have been loading file formats that were specifically built to hold spatial information. But often, point data is stored in table format, with a column for the x coordinates and a column for the y coordinates. The easiest way to get this type of data into a GeoDataFrame is with the `geopandas` function `geopandas.points_from_xy`, which takes list-like sequences of x and y coordinates. In this case, we can get these list-like sequences from columns of a pandas `DataFrame` that we get from `read_csv`.
+This will cut out a smaller area, defined by a box in units of the projection, discarding the rest of the data. The resultant GeoDataframe is found in the `cropfield_crop` object. We can check the total bounds of this new data as before:
+
+~~~
+cropfield_crop.total_bounds
+~~~
+{: .language-python}
+
+~~~
+array([119594.384 , 485036.2543, 135169.9266, 500782.531 ])
+~~~
+{: .output}
+
+We can then save this cropped shapefile for use in future, using the `to_file()` method of our GeoDataFrame object:
 
 ```python
-# we get the projection of the point data from our Canopy Height Model, 
-# after examining the pandas DataFrame and seeing that the CRSs are the same
-import rioxarray
-CHM_HARV <-
-  rioxarray.open("data/NEON-DS-Airborne-Remote-Sensing/HARV/CHM/HARV_chmCrop.tif")
-
-# plotting locations in CRS coordinates using CHM_HARV's CRS
-plot_locations_HARV =
-  pd.read_csv("data/NEON-DS-Site-Layout-Files/HARV/HARV_PlotLocations.csv")
-plot_locations_HARV = gpd.GeoDataFrame(plot_locations_HARV, 
-                    geometry=gpd.points_from_xy(plot_locations_HARV.easting, plot_locations_HARV.northing), 
-                    crs=CHM_HARV.rio.crs)
+cropfield_crop.to_file('cropped_field.shp')
 ```
 
+This will write a new shape file to disk, containing only the data from our cropped area. It can be read in again at a later time using the `read_file()` method we have been using above.
 
 ## Plotting a Shapefile
 
-Any `GeoDataFrame` can be plotted in CRS units to view the shape of the object with `.plot()`.
+We can now plot this data. Any `GeoDataFrame` can be plotted in CRS units to view the shape of the object with `.plot()`.
 
-```{r}
-aoi_boundary_HARV.plot()
+```python
+cropfield_crop.plot()
 ```
 
```

Comment:
Add an output section for plotting?

Reply:
Ah yes, good point. Added an image to the output section for the figure plot.